bokeh/layouts.py

```diff
@@ -335,7 +348,8 @@ def gridplot(children, sizing_mode=None, toolbar_location='above', ncols=None,
     elif toolbar_location == 'right':
         return Row(children=[grid, toolbar], sizing_mode=sizing_mode)
 
-def grid(children=[], sizing_mode=None, nrows=None, ncols=None):
+def grid(children: List[LayoutDOM] | List[List[LayoutDOM]] | Row | Column = [], *,
+        sizing_mode: Optional[SizingMode] = None, nrows: int | None = None, ncols: int | None = None):
```

Comment:
What is the rationale for using different syntaxes here? I would strongly prefer to use `Optional[...]` consistently here and throughout and reserve `foo | bar` only for cases that do not involve None

Reply:
Work in progress. However, `a | b` syntax is supported only starting in Python 3.10, so depending on the context, we will need to use `Union[]` or `Optional[]` (e.g. in type aliases). The new syntax works here, because of lazy annotations.